From de176ee81bb5a9a3319519d062c59a879a61e174 Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Mon, 1 Oct 2018 18:33:32 +0200
Subject: [PATCH 39/48] FingerprintService: add property to prevent cleanup of
 unused fingerprints

Restores Oreo behaviour.

Usage:
    ro.fingerprint.cleanup.unused=false

Change-Id: Id032fae5c6ae70ce57a60c6f5d3dbe0a6cd33258
---
 .../com/android/server/fingerprint/FingerprintService.java     | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/fingerprint/FingerprintService.java b/services/core/java/com/android/server/fingerprint/FingerprintService.java
index 9f9ed33f753..22dc07d1550 100644
--- a/services/core/java/com/android/server/fingerprint/FingerprintService.java
+++ b/services/core/java/com/android/server/fingerprint/FingerprintService.java
@@ -62,6 +62,7 @@ import android.os.RemoteException;
 import android.os.SELinux;
 import android.os.ServiceManager;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.security.KeyStore;
@@ -336,7 +337,7 @@ public class FingerprintService extends SystemService implements IHwBinder.Death
      * @param userId
      */
     private void doFingerprintCleanupForUser(int userId) {
-        if (CLEANUP_UNUSED_FP) {
+        if (SystemProperties.getBoolean("ro.fingerprint.cleanup.unused", CLEANUP_UNUSED_FP)) {
             enumerateUser(userId);
         }
     }
-- 
2.17.1

